src/Core/Authoring/PhysicalPackageFile.cs

```diff
@@ -89,9 +91,171 @@ public IEnumerable<FrameworkName> SupportedFrameworks
 
         public Stream GetStream()
         {
-            return _streamFactory != null ? _streamFactory() : File.OpenRead(SourcePath);
+            try
+            {
+                return _streamFactory != null ? _streamFactory() : File.OpenRead(SourcePath);
+            }
+            catch (PathTooLongException)
+            {
+                return Win32File.OpenRead(SourcePath);
+            }
+        }
+
+        /// <summary>
+        /// From: http://www.codeproject.com/Articles/22013/NET-Workaround-for-PathTooLongException
+        /// </summary>
+        class Win32File

```

Comment:
Put All Native calls in a NativeMethods class, and always prefix classes with their accessors